lib/template-parts.php

```diff
@@ -158,13 +166,32 @@ function filter_rest_wp_template_part_collection_params( $query_params ) {
 apply_filters( 'rest_wp_template_part_collection_params', 'filter_rest_wp_template_part_collection_params', 99, 1 );
 
 /**
- * Filter for supporting the `theme` parameter in `wp_template_part` queries.
+ * Filter for supporting the `resolved`, `template`, and `theme` parameters in `wp_template_part` queries.
  *
  * @param array           $args    The query arguments.
  * @param WP_REST_Request $request The request object.
  * @return array Filtered $args.
  */
 function filter_rest_wp_template_part_query( $args, $request ) {
+	if ( $request['resolved'] ) {
```

Comment:
```suggestion
	if ( $request['resolved'] || $request['template'] ) {
```

Reply:
FTR, I was trying to retain consistency with `lib/templates.php` here, where we only check for `resolved`: https://github.com/WordPress/gutenberg/blob/206c5f12126505bff374eef4ad137a0f267920cb/lib/templates.php#L184

...but also take the template `slug` into account:

https://github.com/WordPress/gutenberg/blob/206c5f12126505bff374eef4ad137a0f267920cb/lib/templates.php#L186

The difference is that in the case of templates, we can use the `slug` field that already exists (as part of the entities endpoint), so we need the `resolved` flag to convey the different semantics (only return 'resolved' templates that match the `slug` vs return _all_ templates that match it (e.g. including all auto-drafts)).

---

We _could_ theoretically assign a different meaning to a template parts query that has a `template` arg but not a `resolved` one -- it'd be all template parts for all templates matching that `template` slug (including auto-drafts etc) -- but I doubt that'd be very useful in practice.

Reply:
I'll apply this change and add a comment with the gist of the above.

Reply:
For template parts, `template` implies `resolved`. I can't think of a case where the opposite would be useful.